flake.nix

```diff
@@ -74,8 +74,20 @@
         ${getExe pkgs.reuse} --suppress-deprecation lint
         ${getExe pkgs.lychee} --offline --no-progress .
         ${getExe lint-python}
+        ${getExe format-cpp}
       '';
 
+      format-cpp = pkgs.writeShellApplication {
+        name = "format-cpp";
+        runtimeInputs = [pkgs.clang-tools_18];
+        text = ''
+          set +u
+          ARG="$1"
+          [ -z "$1" ] && ARG="-n"
+          find sw -path sw/cheri/build -prune -o \( -name "*.c" -o -name "*.cc" -o -name "*.h" -o -name "*.hh" \) -exec clang-format "$ARG" {} \;
```

Comment:
The exclude path is a little hidden here. Just a note, I won't ask you to change this.